From c3a75cc1dec322173b9a4314d12550bde09a9871 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Wed, 4 Oct 2023 08:23:20 +0200
Subject: [PATCH] palmGetResource: Fix loading of local resources

After WAM/Chromium implemented Security Origin with https://github.com/webosose/chromium94/commit/141f7ae66bd51e7c6574133aab7a268d931e9aad#diff-864005cba641bdb78feeb85fa1c212bde631ec799541c75e9127370d847ab31f and https://github.com/webosose/wam/commit/977ea6b0aa28aff4592c91fb6cdc9b83f8aa9e40 non aware frameworks such as Enyo cannot find their paths anymore due to the prefix. Fix this by removing the security origin (which consists of "appid" suffixed with "-webos" from the path before trying to load it.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
Upstream-Status: Pending

 .../injection/renderer/resources/webossystem_injection.js    | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/neva/injection/renderer/resources/webossystem_injection.js b/src/neva/injection/renderer/resources/webossystem_injection.js
index a4add503e8..cb7968bb72 100644
--- a/src/neva/injection/renderer/resources/webossystem_injection.js
+++ b/src/neva/injection/renderer/resources/webossystem_injection.js
@@ -18,6 +18,11 @@ window.webOSGetResource = function(p1, p2) {
   var a = p1 || "";
   var b = p2 || "";
 
+  if(a.indexOf("-webos/") !== -1){
+      var regex = /(file:\/\/)([a-z\.\-0-9]{1,}-webos)(\/[a-z0-9\.\-\/]{1,})/;
+      a = a.replace(regex, "$1$3");
+  }
+
   if(a.indexOf("file://") === 0){
       a = a.substring(7);
   }
